plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -41,19 +46,75 @@ func NewStrategyStore(options Options, logger *zap.Logger) (ss.StrategyStore, er
 		logger:            logger,
 		serviceStrategies: make(map[string]*sampling.SamplingStrategyResponse),
 	}
+
+	// Read strategies
 	strategies, err := loadStrategies(options.StrategiesFile)
 	if err != nil {
 		return nil, err
 	}
 	h.parseStrategies(strategies)
+
+	// Watch strategies file for changes.
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logger.Error("failed to create a new watcher for the sampling strategies file", zap.Error(err))
+	}
+
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				if event.Op&fsnotify.Remove == fsnotify.Remove {
+					logger.Warn("the sampling strategies file has been removed")
+
+					watcher.Remove(event.Name)
+					watcher.Add(options.StrategiesFile)
```

Comment:
below we're checking for error, but not here?

also, wouldn't this always fail since the file was removed? Perhaps Remove should be just no-op, aside from logging.

Reply:
> below we're checking for error, but not here?

Will add error checking.

> also, wouldn't this always fail since the file was removed? 

Please check comment below.